File: BarcodeScan/app/src/main/java/com/example/mybarcodescanner/MainActivity.java
package com.example.mybarcodescanner;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.google.zxing.integration.android.IntentIntegrator;
import com.google.zxing.integration.android.IntentResult;

@SuppressWarnings("deprecation")
public class MainActivity extends AppCompatActivity implements View.OnClickListener{

    Button scan_button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        scan_button = findViewById(R.id.scan_button);
        scan_button.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {

        scanCode();
    }

    @Deprecated
    private void scanCode() {
        IntentIntegrator integrator = new IntentIntegrator(this);
        integrator.setCaptureActivity(CaptureAct.class);
        integrator.setOrientationLocked(false);
        integrator.setDesiredBarcodeFormats(IntentIntegrator.ALL_CODE_TYPES);
        integrator.setPrompt("Scanning Barcode of Item");
        integrator.initiateScan();

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        IntentResult result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data);
        if (result != null) {
            if (result.getContents() != null) {
                AlertDialog.Builder builder = new AlertDialog.Builder(this);
                builder.setMessage("Item " + result.getContents() + " added.");
                builder.setTitle("Scanning result");
                builder.setPositiveButton("Scan New Item", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        scanCode();
                    }
                }).setNegativeButton("finish", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        finish();
                    }
                });
                AlertDialog dialog = builder.create();
                dialog.show();
            } else {
                Toast.makeText(this, "No results", Toast.LENGTH_LONG).show();
            }
        } else {
            super.onActivityResult(requestCode, resultCode, data);
        }

    }
}

File: RecipeRecommendation/app/src/main/java/com/example/reciperecommendation/MainActivity.kt
package com.example.reciperecommendation

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var recipeName: TextView = findViewById(R.id.recipeName)
        val button = findViewById<Button>(R.id.button)
        var name: String = "Apricot chicken with chickpeas and sweet potato"

        button.setOnClickListener({
            //val i = Intent(Intent.ACTION_VIEW, Uri.parse("https://www.taste.com.au/recipes/apricot-chicken-chickpeas-sweet-potato/f2bcbc0c-30d5-4a68-bbd3-d1993e662acd?r=recipes/lowfatrecipes&c=53b15009-0732-4041-8225-4a02912b67c5/Low%20fat%20recipes" ))
            val uri = Uri.parse("https://www.taste.com.au/recipes/apricot-chicken-chickpeas-sweet-potato/f2bcbc0c-30d5-4a68-bbd3-d1993e662acd?r=recipes/lowfatrecipes&c=53b15009-0732-4041-8225-4a02912b67c5/Low%20fat%20recipes")

            val intent = Intent(Intent.ACTION_VIEW, uri)
            startActivity(intent)

        })
    }
}